### PR TITLE
[Enhancement](HttpServer) Add http interface authentication for BE

### LINKED ISSUE #15658: [Enhancement] add http interface authentication and support https
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, doris supports access to fe / be only through http protocol, not https protocol. In addition, http does not perform interface authentication, which may cause security problems.

### Solution

Add http interface authentication and support https for doris, eliminate potential security risks and strengthen enterprise-level security support.
[Support https protocol and interface authentication](https://cwiki.apache.org/confluence/display/DORIS/DSIP-029%3A+Support+https+protocol+and+interface+authentication)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)]

### PR DESCRIPTION
**Pull request by xiaojunjie**
# Proposed changes

Issue Number: close #15658

## Problem summary

1. Organize http documents
2. Add http interface authentication for FE
3. Support https interface for FE
4. Provide authentication interface
5. **Add http interface authentication for BE**
6. Support https interface for BE

### Interface with authentication

- [x] check_rpc_channel   (**admin**)
- [x] checksum   (**admin**)
- [x] check_tablet_segment   (**admin**)
- [x] compaction   (**admin**)
- [ ] config   (**none** for FE)
- [ ] download   (**already** authenticated)
- [ ] health   (**none** for monitor)
- [ ] jeprofile   (**none** for web)
- [x] meta   (**admin**)
- [x] metrics   (**none** for prometheus)
- [ ] monitor   (_deleted_)
- [x] pad_rowset   (**admin**)
- [ ] pprof   (**none** for web)
- [x] reload_tablet   (**admin**)
- [x] reset_rpc_channel   (**admin**)
- [x] restore_tablet   (**admin**)
- [x] snapshot   (**admin**)
- [ ] stream_load_2pc (**already** authenticated)
- [ ] stream_load (**already** authenticated)
- [x] tablet_migration   (**admin**)
- [x] tablets_distribution   (**admin**)
- [x] tablets_info   (**admin**)
- [x] version   (**none**)

## Checklist(Required)

* [x] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.96 seconds
 stream load tsv:          448 seconds loaded 74807831229 Bytes, about 159 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          73 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230314084729_clickbench_pr_113955.html

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run p0

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run FE UT

**Comment by xiaojunjie** [comment]
run buildall

**Comment by yangzhg** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run p0

**Review by yangzhg — DISMISSED** [review]
LGTM

**Comment by xiaojunjie** [comment]
run buildall

**Review by yangzhg — DISMISSED** [review]
LGTM

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

File: be/src/service/http_service.cpp [hunk 1]
```diff
@@ -129,42 +132,51 @@ Status HttpService::start() {
...
+    checksum_action->auth(TPrivilegeHier::GLOBAL, TPrivilegeType::ADMIN);
     _ev_http_server->register_handler(HttpMethod::GET, "/api/checksum", checksum_action);
 
     // Register BE reload tablet action
     ReloadTabletAction* reload_tablet_action = _pool.add(new ReloadTabletAction(_env));
+    reload_tablet_action->auth(TPrivilegeHier::GLOBAL, TPrivilegeType::ADMIN);
     _ev_http_server->register_handler(HttpMethod::GET, "/api/reload_tablet", reload_tablet_action);
 
     RestoreTabletAction* restore_tablet_action = _pool.add(new RestoreTabletAction(_env));
+    restore_tablet_action->auth(TPrivilegeHier::GLOBAL, TPrivilegeType::ADMIN);
```
Comment (morningman) [inline be/src/service/http_service.cpp:156]: I think this can be done in the constructor of each action.

File: be/src/common/config.h [hunk 2]
```diff
@@ -375,6 +375,8 @@ CONF_Bool(enable_https, "false");
 CONF_String(ssl_certificate_path, "");
 // Path of private key
 CONF_String(ssl_private_key_path, "");
+// Whether to check authorization
+CONF_Bool(enable_auth, "false");
```
Comment (morningman) [inline be/src/common/config.h:379]:
```suggestion
CONF_Bool(enable_http_auth, "false");
```

File: be/src/http/action/check_rpc_channel_action.h [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
 
-#include "http/http_handler.h"
+#include "http/http_handler_with_auth.h"
 
 namespace doris {
 class ExecEnv;
 class HttpRequest;
 
-class CheckRPCChannelAction : public HttpHandler {
+class CheckRPCChannelAction : public HttpHandlerWithAuth {
```
Comment (morningman) [inline be/src/http/action/check_rpc_channel_action.h:26]: This will be called from FE, which may not contain auth info

**Comment by xiaojunjie** [comment]
run buildall

**Comment by xiaojunjie** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**